### PR TITLE
cringecrafting to slapcrafting conversion

### PR DESCRIPTION
**Pull request by Spyroshark**
## Changelog

:cl:
add: Added new mechanics or gameplay changes
add: Added more things
del: Removed old things
qol: made something easier to use
balance: rebalanced something
fix: fixed a few things
soundadd: added a new sound thingy
sounddel: removed an old sound thingy
imageadd: added some icons and images
imagedel: deleted some icons and images
spellcheck: fixed a few typos
code: changed some code
refactor: refactored some code
config: changed some config setting
admin: messed with admin stuff
server: something server ops should know
/:cl:

### CODE REVIEW

File: code/modules/slapcrafting/recipes/components/metal_shaping.dm [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	examine_hint = "You could cut the metal into rods with a welder..."
+	category = SLAP_CAT_COMPONENTS
+	steps = list(
+		/datum/slapcraft_step/stack/iron/one,
+		/datum/slapcraft_step/tool/welder
+	)
+	result_type = /obj/item/stack/rods
+
+/datum/slapcraft_recipe/metal_rods/create_item(item_path, obj/item/slapcraft_assembly/assembly)
+	return new item_path(assembly.drop_location(), 2)
```
Comment (Kapu1178) [inline code/modules/slapcrafting/recipes/components/metal_shaping.dm]: make a new typedef for `rods/two` and just spawn that instead

File: code/modules/slapcrafting/base_steps/slapcraft_step_stack.dm [hunk 2]
```diff
@@ -99,7 +99,18 @@
 	finish_msg = "%USER% inserts a rod to the %TARGET%."
 	finish_msg_self = "You insert a rod to the %TARGET%."
 
-/datum/slapcraft_step/stack/metal/one
+/datum/slapcraft_step/stack/rod/two
+	desc = "Add 2 rod to the assembly."
```
Comment (Kapu1178) [inline code/modules/slapcrafting/base_steps/slapcraft_step_stack.dm]: rods

**Review by Kapu1178 — CHANGES_REQUESTED** [review]

File: code/game/objects/items/weaponry.dm [hunk 3]
```diff
@@ -944,3 +897,23 @@ for further reading, please see: https://github.com/tgstation/tgstation/pull/301
...
 	return ..()
+
+/obj/item/mace
+	name = "\improper iron mace"
+	desc = "A crude mace made made from pieces of metal welded together."
+	icon_state = "shitty_mace"
+	inhand_icon_state = "mace"
+	lefthand_file = 'icons/mob/inhands/weapons/melee_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/melee_righthand.dmi'
+	force = 2
```
Comment (Kapu1178) [inline code/game/objects/items/weaponry.dm]: only 2...?

File: code/game/objects/items/weaponry.dm [hunk 4]
```diff
@@ -944,3 +897,23 @@ for further reading, please see: https://github.com/tgstation/tgstation/pull/301
 		balloon_alert(user, "you're too weak!")
 		return
 	return ..()
+
+/obj/item/mace
+	name = "\improper iron mace"
```
Comment (Kapu1178) [inline code/game/objects/items/weaponry.dm]: lowercase names are improper nouns already

File: code/modules/slapcrafting/base_steps/slapcraft_step_weapon.dm [hunk 5]
```diff
@@ -5,7 +5,7 @@
 	check_types = FALSE
 	check_if_mob_can_drop_item = FALSE
 
-	list_desc = "sharp implement"
+	list_desc = "cutting implement"
```
Comment (Kapu1178) [inline code/modules/slapcrafting/base_steps/slapcraft_step_weapon.dm]: y

File: code/modules/slapcrafting/base_steps/slapcraft_step_weapon.dm [hunk 6]
```diff
@@ -39,14 +39,18 @@
 		playsound(assembly, item.hitsound, 50, TRUE, -1)
 
 /datum/slapcraft_step/attack/sharp
+	desc = "Cut the assembly with somethign sharp." //this is a generic description and should be overriden
```
Comment (Kapu1178) [inline code/modules/slapcrafting/base_steps/slapcraft_step_weapon.dm]:
```suggestion
	desc = "Cut the assembly with something sharp." //this is a generic description and should be overridden
```

File: code/modules/slapcrafting/recipes/weapons/melee.dm [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+/datum/slapcraft_step/spear
+	desc = "Start with a spear."
+	item_types = list(/obj/item/spear)
+
+/datum/slapcraft_recipe/explosive_lance/create_item(item_path, obj/item/slapcraft_assembly/assembly)
+	var/obj/item/spear/explosive/spear = new item_path(assembly.drop_location())
+	if(!spear)
+		return
+	var/obj/item/grenade/G = assembly.contents.Find(/obj/item/grenade)
```
Comment (Kapu1178) [inline code/modules/slapcrafting/recipes/weapons/melee.dm]:
```suggestion
	var/obj/item/grenade/G = locate() in assembly
```

File: code/modules/slapcrafting/recipes/weapons/melee.dm [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+	)
+	result_type = /obj/item/spear/explosive
+
+/datum/slapcraft_step/spear
+	desc = "Start with a spear."
+	item_types = list(/obj/item/spear)
+
+/datum/slapcraft_recipe/explosive_lance/create_item(item_path, obj/item/slapcraft_assembly/assembly)
+	var/obj/item/spear/explosive/spear = new item_path(assembly.drop_location())
+	if(!spear)
```
Comment (Kapu1178) [inline code/modules/slapcrafting/recipes/weapons/melee.dm]: This can't happen.

**Review by Kapu1178 — CHANGES_REQUESTED** [review]
I'd prefer if we avoided stuff like `/datum/slapcraft_step/item/pipe/add` because iirc it increases time to lookup recipes and shit.

Also, step datums should be moved to their respective file, otherwise it's going to be impossible to find them. Exception: Hyper-specific types.

File: code/modules/slapcrafting/base_steps/slapcraft_step_weapon.dm (hunk 5 above)
Comment (Spyroshark) [reply code/modules/slapcrafting/base_steps/slapcraft_step_weapon.dm]: "cutting implement" sounds more natural then "sharp implement" to me. cutting is what the implement is for. i'll change it if you want but imo it makes sense

**Pull request merged**